clients/src/main/java/org/apache/kafka/clients/producer/internals/FutureRecordMetadata.java

```diff
@@ -72,6 +78,14 @@ public long timestamp() {
         return this.timestamp;
     }
 
+    public long checksum() {
+        return this.checksum;
+    }
+
+    public int size() {
```

Comment:
Yes that would remove all the confusion what the size is. Then, I will have keySize and valueSize in both RecordMetadata and ConsumerRecord, instead of size.